homeassistant/helpers/trigger.py

```diff
@@ -493,6 +553,71 @@ async def with_vars(
     return wrapper_func
 
 
+async def _async_attach_trigger_cls(
+    hass: HomeAssistant,
+    trigger_cls: type[Trigger],
+    trigger_key: str,
+    conf: ConfigType,
+    action: Callable,
+    trigger_info: TriggerInfo,
+) -> CALLBACK_TYPE:
+    """Initialize a new Trigger class and attach it."""
+
+    def action_payload_builder(
+        extra_trigger_payload: dict[str, Any], description: str
+    ) -> dict[str, Any]:
+        """Build action variables."""
+        payload = {
+            "trigger": {
+                **trigger_info["trigger_data"],
+                CONF_PLATFORM: trigger_key,
+                "description": description,
+                **extra_trigger_payload,
+            }
+        }
+        if CONF_VARIABLES in conf:
+            trigger_variables = conf[CONF_VARIABLES]
+            payload.update(trigger_variables.async_render(hass, payload))
+        return payload
+
+    # Wrap sync action so that it is always async.
+    # This should be removed when sync actions are no longer supported.
+    match get_hassjob_callable_job_type(action):
+        case HassJobType.Executor:
+            original_action = action
```

Comment:
We should call `homeassistant.helpers.frame.report_usage` and say this will stop working, both here and in the old trigger API.

Reply:
Lets leave the deprecation of the sync-actions to a separate PR.